src/main/java/org/ibp/api/java/impl/middleware/design/generator/ExperimentDesignGenerator.java

```diff
@@ -0,0 +1,677 @@
+import java.util.Random;
+import java.util.Set;
+import java.util.StringTokenizer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+public class ExperimentDesignGenerator {
+
+	public static final String NCLATIN_PARAM = "nclatin";
+	public static final String NRLATIN_PARAM = "nrlatin";
+	public static final String REPLATINGROUPS_PARAM = "replatingroups";
+	public static final String COLUMNFACTOR_PARAM = "columnfactor";
+	public static final String ROWFACTOR_PARAM = "rowfactor";
+	public static final String NCOLUMNS_PARAM = "ncolumns";
+	public static final String NROWS_PARAM = "nrows";
+	public static final String NBLATIN_PARAM = "nblatin";
+	public static final String REPLICATEFACTOR_PARAM = "replicatefactor";
+	public static final String TREATMENTFACTOR_PARAM = "treatmentfactor";
+	public static final String INITIAL_TREATMENT_NUMBER_PARAM = "initialtreatnum";
+	public static final String NREPLICATES_PARAM = "nreplicates";
+	public static final String NTREATMENTS_PARAM = "ntreatments";
+	public static final String BLOCKSIZE_PARAM = "blocksize";
+	public static final String TIMELIMIT_PARAM = "timelimit";
+	public static final String LEVELS_PARAM = "levels";
+	public static final String TREATMENTFACTORS_PARAM = "treatmentfactors";
+	public static final String PLOTFACTOR_PARAM = "plotfactor";
+	public static final String INITIAL_PLOT_NUMBER_PARAM = "initialplotnum";
+	public static final String BLOCKFACTOR_PARAM = "blockfactor";
+	public static final String NBLOCKS_PARAM = "nblocks";
+	public static final String OUTPUTFILE_PARAM = "outputfile";
+	public static final String SEED_PARAM = "seed";
+	public static final String NCONTROLS_PARAM = "ncontrols";
+	public static final String NUMBER_TRIALS_PARAM = "numbertrials";
+	public static final String NREPEATS_PARAM = "nrepeats";
+
+	public static final String RANDOMIZED_COMPLETE_BLOCK_DESIGN = "RandomizedBlock";
+	static final String RESOLVABLE_INCOMPLETE_BLOCK_DESIGN = "ResolvableIncompleteBlock";
+	public static final String RESOLVABLE_ROW_COL_DESIGN = "ResolvableRowColumn";
+	public static final String AUGMENTED_RANDOMIZED_BLOCK_DESIGN = "Augmented";
+	public static final String P_REP_DESIGN = "Prep";
+
+	private static final int STARTING_ENTRY_NUMBER = 1;
+
+	private static final Logger LOG = LoggerFactory.getLogger(ExperimentDesignGenerator.class);
+	private static final List<Integer> EXP_DESIGN_VARIABLE_IDS =
+		Arrays.asList(TermId.PLOT_NO.getId(), TermId.REP_NO.getId(), TermId.BLOCK_NO.getId(), TermId.ROW.getId(), TermId.COL.getId());
+
+	@Resource
+	private DesignRunner designRunner;
+
+	@Resource
+	private StudyDataManager studyDataManager;
+
+	@Resource
+	private DatasetService datasetService;
+
+	@Resource
+	private OntologyDataManager ontologyDataManager;
+
+	@Resource
+	private StandardVariableTransformer standardVariableTransformer;
+
+	private final Random random = new Random();
+
+	public MainDesign createRandomizedCompleteBlockDesign(
```

Comment:
Kind of weird how this is implemented. In only one class we do have createXXXDesign without following any rule and not taking advantage of polymorphism as we have in another areas of experimental design generation. Something we can improve

Reply:
Yes I agree, will create a tech backlog ticket for this.